pennylane/measurements.py

```diff
@@ -125,6 +152,121 @@ def __init__(self, return_type, obs=None, wires=None, eigvals=None, id=None):
         # Queue the measurement process
         self.queue()
 
+    def shape(self, device=None):
+        """The expected output shape of the MeasurementProcess.
+
+        Note that the output shape is dependent on the device when:
```

Comment:
Iirc, the number of qubits would be dynamically allocated, right? The number of all qubits would be required in the `qml.state` case. Otherwise, we're extracting other information from the device, like:
* `cutoff` value;
* shots;
* shot vector.